### PR TITLE
Add a `bindir` parameter to build.sh.

### PR DESCRIPTION
**Pull request by pgavlin**
This parameter can be used to redirect the usual output path from
`./bin` to a different directory.

### CODE REVIEW

**Comment by pgavlin** [comment]
@ellismg @joperezr PTAL

File: build.sh [hunk 1]
```diff
@@ -583,6 +583,16 @@ while :; do
             fi
             ;;
 
+        bindir)
+            if [ -n "$2" ]; then
+                __RootBinDir="$(readlink -e "$2")"
```
Comment (ellismg) [inline build.sh]: -e does not work in OSX, and presumably other BSDs.

**Comment by joperezr** [comment]
Will CMAKE_BINARY_DIR match $__IntermediatesDir in this case? I ask because in order for versioning to work, we need [this variable](https://github.com/dotnet/coreclr/blob/master/CMakeLists.txt#L18) to match [this one on the script](https://github.com/dotnet/coreclr/blob/master/build.sh#L151)

**Comment by BruceForstall** [comment]
Are you going to add an equivalent option to build.cmd?

**Comment by pgavlin** [comment]
@BruceForstall if you think it would be useful, yes, I can.

**Comment by BruceForstall** [comment]
Well, I don't know how useful it would be (I don't know of any pressing scenarios). It was more a question of whether parity was important.

**Comment by pgavlin** [comment]
I've been using it to separate artifacts built targeting Fedora from artifacts built targeting Ubuntu. I can imagine that it might also be useful in order to separate build artifacts for one branch from build artifacts from another, which is a scenario that could arise on any platform. I'll add it with a follow-up change.

**Comment by BruceForstall** [comment]
I mentioned that latter scenario (different binaries for different branches) to Russ Hadley yesterday as a way to separate "master" and "feature" branches for the purpose of JIT asm diffs, by not trashing the previous built artifacts.

**Comment by pgavlin** [comment]
@joperezr yes, the two will match.

**Comment by joperezr** [comment]
ok, then it LGTM.

**Pull request merged**